### ISSUE TITLE
Upgrade Faraday to 2.X

### ISSUE THREAD

**Issue by j-monteiro** [phrase/phrase-ota-i18n#20]
Hi! After installing this gem I've noticed 2 depreaction warnings when running the app server or rspec.

```bash
Top level ::CompositeIO is deprecated, require 'multipart/post' and use `Multipart::Post::CompositeReadIO` instead!
Top level ::Parts is deprecated, require 'multipart/post' and use `Multipart::Post::Parts` instead!
````

After some research I've reached the conclusion that this is due to the faraday gem that is being forced to stay at 1.X due to the `faraday_middleware` dependency. The later is currently deprecated and should not be used as stated in the [project README](https://github.com/lostisland/faraday_middleware/blob/main/README.md#%EF%B8%8F-deprecation-warning-%EF%B8%8F).

Would be nice to replace this dependency with the recommentded [standalone replacement](https://github.com/tisba/faraday-follow-redirects)

**Comment by j-monteiro** [comment]
By updating the faraday gem with `bundle update faraday` it updated to 1.10.3 and this version already fixes the warnings, nonetheless it would be cool to upgrade to faraday 2 if you guys agree. I'm willing to submit a PR with that.

**Comment by theSoenke** [comment]
Hi @j-monteiro, thanks for raising this! Also had this on my list but haven't started working on the upgrade.  If you would like to submit a PR for this, that would be very welcome.

**Comment by theSoenke** [comment]
It's now released as version 1.0.3. Thanks again for the fix!
